### PR TITLE
Fixes NPE from uncached users

### PR DESCRIPTION
**Pull request by oliver276**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [x] Internal code
- [ ] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

## Description

Fixes NPE from uncached users in private channel events

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java [hunk 1]
```diff
@@ -1363,16 +1363,25 @@ public ReceivedMessage createMessage(DataObject jsonObject, @Nullable MessageCha
...
-        if (channel == null && jsonObject.isNull("guild_id"))
+        if (jsonObject.isNull("guild_id"))
         {
-            DataObject channelData = DataObject.empty()
-                    .put("id", channelId);
-            //if we see an author that isn't us, we can assume that is the other side of this private channel
-            //if the author is us, we learn no information about the user at the other end
-            if (authorId != getJDA().getSelfUser().getIdLong())
+            //we know it's a private channel
+            PrivateChannelImpl priv = (PrivateChannelImpl) channel;
```
Comment (Andre601) [inline src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java:1369]: Wouldn't this cause an exception when `channel` is null?

Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java:1369]: No.

Comment (Andre601) [reply src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java:1369]: Are you sure?
To my knowledge does casting a null object cause an exception. And unless I miss something is `channel` nullable, no?

Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java:1369]:
```
jshell> (String) null
$1 ==> null
```

**Review by DV8FromTheWorld — APPROVED** [review]

**Review by DV8FromTheWorld — APPROVED** [review]

**Pull request merged**